bluetooth/src/iosMain/kotlin/scanner/Scanner.kt

```diff
@@ -27,47 +27,43 @@ actual class Scanner internal constructor(autoEnableBluetooth: Boolean,
     }
 
     @Suppress("CONFLICTING_OVERLOADS")
-    private class CentralManagerDelegate(val bluetoothScanner: Scanner) : NSObject(), CBCentralManagerDelegateProtocol {
+    private val centralManagerDelegate = object : NSObject(), CBCentralManagerDelegateProtocol {
 
         override fun centralManager(central: CBCentralManager, didDiscoverPeripheral: CBPeripheral, advertisementData: Map<Any?, *>, RSSI: NSNumber) {
             super.centralManager(central, didDiscoverPeripheral, advertisementData, RSSI)
 
-            bluetoothScanner.discoverPeripheral(central, didDiscoverPeripheral, advertisementData.typedMap(), RSSI.intValue)
+            discoverPeripheral(central, didDiscoverPeripheral, advertisementData.typedMap(), RSSI.intValue)
         }
 
         override fun centralManagerDidUpdateState(central: CBCentralManager) {
             when (central.state) {
-                CBCentralManagerStatePoweredOn -> bluetoothScanner.bluetoothEnabled()
-                else -> bluetoothScanner.bluetoothDisabled()
+                CBCentralManagerStatePoweredOn -> bluetoothEnabled()
+                else -> bluetoothDisabled()
             }
         }
 
         override fun centralManager(central: CBCentralManager, didConnectPeripheral: CBPeripheral) {
             super.centralManager(central, didConnectPeripheral)
 
-            bluetoothScanner.connectionManagerMap[didConnectPeripheral.identifier]?.let {
-                it.didConnect()
-            }
+            val connectionManager = connectionManagerMap[didConnectPeripheral.identifier] ?: return
```

Comment:
Why not just `connectionManagerMap[didConnectPeripheral.identifier]?.didConnect()`?